### PR TITLE
Upgrade to vanilla 3.0 and react-components 0.37

### LINKED ISSUE #1248: Upgrade to Vanilla 3
Vanilla 3 includes breaking changes so this update will need to be done with plenty of QA: https://vanillaframework.io/docs/migration-guide-to-v3

### PR DESCRIPTION
**Pull request by goulinkh**
## Done

- Upgrade `vanilla-framework` and `@canonical/react-components` to latest versions
- Simplify some styles
- Add breakpoint helper: the new version of Vanilla introduces the removal of the breakpoint `medium`,  a way to solve this issue is by redifining the value in the project but I feel that this will be like fighting against the flow so Instead I followed the recommended way in the docs `small < ... < large`
- Fix models row on hover issue
- Fix the WebCLI width issue: https://github.com/canonical/jaas-dashboard/issues/1241#issuecomment-1250568218

## QA

- Run the dashboard locally
- Go through all the views and popups
- Compare each UI element with the existing dashboard one

## Details

Fixes #1248, #1244

### CODE REVIEW

File: src/components/Aside/Aside.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import classnames from "classnames";
 import SlideInOut from "animations/SlideInOut";
+import classnames from "classnames";
 
-import Spinner from "@canonical/react-components/dist/components/Spinner/Spinner";
+import { Spinner } from "@canonical/react-components";
```
Comment (goulinkh) [inline src/components/Aside/Aside.tsx:4]: Replaced all the imports of this type from `@canonical/react-components`

File: src/components/Aside/_aside.scss [hunk 2]
```diff
@@ -1,16 +1,7 @@
 @import "vanilla-framework/scss/vanilla";
+@import "../../scss/breakpoints";
 
 .l-aside {
-  min-height: 100vh;
-  overflow-y: scroll !important;
-  padding: 0;
-  position: relative;
-
-  .p-panel {
```
Comment (goulinkh) [inline src/components/Aside/_aside.scss:9]: Removing these lines didn't affect the behavior of this element

File: src/components/ChipGroup/_chip-group.scss [hunk 3]
```diff
@@ -1,14 +1,19 @@
 @import "vanilla-framework/scss/vanilla";
 
 .chip-group {
-  margin-top: -0.4rem;
-  padding-bottom: 0.5rem;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  margin: 1.25rem 0;
```
Comment (goulinkh) [inline src/components/ChipGroup/_chip-group.scss:7]: the `chip-group` spacing `margin`/`padding` was overridden in different places, I got rid of the overrides and fixed the styling here.
| units | models |
:-------------------------:|:-------------------------:
image | image

File: src/components/ConfirmationModal/_confirmation-modal.scss [hunk 4]
```diff
@@ -48,7 +48,7 @@
   }
 
   &__info-group {
-    margin-bottom: $spv-outer--medium;
+    margin-bottom: $spv--large;
```
Comment (goulinkh) [inline src/components/ConfirmationModal/_confirmation-modal.scss:51]: `$spv-outer--*` are removed, the value of `$spv-outer--medium` matches the value of `$spv--large`

File: src/components/Filter/Filter.js [hunk 5]
```diff
@@ -1,57 +0,0 @@
-import Button from "components/Button/Button";
```
Comment (goulinkh) [inline src/components/Filter/Filter.js:1]: The Filter component isn't used by any view, so I deleted it.

File: src/components/ModelTableList/CloudGroup.js [hunk 6]
```diff
@@ -175,5 +175,5 @@ export default function CloudGroup({ filters }) {
       />
     );
   }
-  return <div className="cloud-group u-overflow--scroll">{cloudTables}</div>;
```
Comment (goulinkh) [inline src/components/ModelTableList/CloudGroup.js:178]: Remove the scrollbar in no mac interface when it's not needed:
image

File: src/components/ModelTableList/OwnerGroup.js [hunk 7]
```diff
@@ -1,22 +1,22 @@
-import { useSelector } from "react-redux";
-import MainTable from "@canonical/react-components/dist/components/MainTable";
-import { useQueryParams, StringParam, withDefault } from "use-query-params";
+import { MainTable } from "@canonical/react-components";
```
Comment (goulinkh) [inline src/components/ModelTableList/OwnerGroup.js:1]: Organized imports

File: src/components/PrimaryNav/PrimaryNav.js [hunk 8]
```diff
@@ -75,7 +76,7 @@ const PrimaryNav = () => {
         <ul className="p-list">
           <li className="p-list__item">
             <span className="version">Version {appVersion}</span>
-            <span className="p-label--new">Beta</span>
+            <StatusLabel appearance="positive">Beta</StatusLabel>
```
Comment (goulinkh) [inline src/components/PrimaryNav/PrimaryNav.js:79]: Replaced the custom component with `StatusLabel`

File: src/layout/BaseLayout/_base-layout.scss [hunk 9]
```diff
@@ -79,47 +76,12 @@
...
 
-  @media (max-width: $breakpoint-medium - 1) {
+  @include mobile {
     &[data-collapsed="true"] {
       transform: translateX(-100%);
     }
   }
 
-  // Override Vanilla to always collapse navigation sidebar
-  @media (min-width: $breakpoint-large - 1) {
```
Comment (goulinkh) [inline src/layout/BaseLayout/_base-layout.scss:89]: Simplify this by simply setting the Vanilla variable `$application-layout--breakpoint-side-nav-expanded: -1` this way the expand style will never get applied.

File: src/pages/EntityDetails/App/App.tsx [hunk 10]
```diff
@@ -1,40 +1,39 @@
+import { Button, MainTable } from "@canonical/react-components";
```
Comment (goulinkh) [inline src/pages/EntityDetails/App/App.tsx:1]: Organize imports.

File: src/pages/Settings/Settings.js [hunk 11]
```diff
@@ -25,18 +26,13 @@ export default function Settings() {
...
-                  type="checkbox"
-                  className="p-switch"
-                  defaultChecked={disableAnalytics}
-                  onChange={() => {
-                    setDisableAnalytics(!disableAnalytics);
-                  }}
-                />
-                <div className="p-switch__slider"></div>
-              </label>
+              <Switch
```
Comment (goulinkh) [inline src/pages/Settings/Settings.js:29]: Replace custom component with `Switch`

File: src/scss/_breakpoints.scss [hunk 12]
```diff
@@ -0,0 +1,36 @@
+@import "settings";
+
```
Comment (goulinkh) [inline src/scss/_breakpoints.scss:2]: Make the media queries more readable by following the [Landon Schropp’s technique](https://css-tricks.com/approaches-media-queries-sass/#aa-landon-schropps-technique) (discovered this by chance :smile:)

Comment (huwshimi) [reply src/scss/_breakpoints.scss:2]: Maybe we should suggest to this to Vanilla.

**Review by huwshimi — APPROVED** [review]
Great work @goulinkh! Just a few QA things I spotted:

1. Side nav is always collapsed instead of open on the controllers and model list pages.

2. Model list status chips need some margins:

Screen Shot 2022-10-13 at 10 09 32 am

3. The access panel radio buttons are a bit out of alignment:

Screen Shot 2022-10-13 at 11 29 30 am

4. If you expand the user in the sidebar and then collapse the sidebar the user stays expanded (previously the user would return to just the icon):

Screen Shot 2022-10-13 at 11 32 46 am

**Comment by goulinkh** [comment]
Thank you so much for reviewing this @huwshimi it's a very annoying one to review as it envloves a lot of energy.
I fixed the the previous problems that you mentioned expect number 3 where it involved reformatting some components code so I was thinking of fixing it in a separate PR: https://github.com/canonical/jaas-dashboard/issues/1194

This PR may introduce some other bugs that we didn't notice so we can just fix them later if you haven't any other issue so far.

**Comment by huwshimi** [comment]
Looking great! One other thing I just noticed is that the owner button loses right padding on medium sizes screens:

Screen Shot 2022-10-14 at 9 30 01 am

**Comment by goulinkh** [comment]
Well seen @huwshimi ! I fixed buttons group issue, merging this and let's keep fixing and getting rid of duplicated stuff with Vanilla and react components while working on other things!

**Pull request merged**